Update docs for testing Fleet in Rancher (#2966)
rancher/fleet: dev/README.md

- kubectl
- ...

### For Mac users

> On Mac OS, Docker runs in a virtual machine; this means that <TODO insert more explanations here>

Reviewer: ```suggestion
On Mac OS, Docker runs in a virtual machine. This means the network setup is more complicated.
```
